changeset.go

```diff
@@ -87,9 +97,14 @@ func (c *Changeset) Apply(doc *rel.Document, mut *rel.Mutation) {
 				c.applyAssocMany(doc, field, mut, v)
 			}
 		default:
-			if field != pField && scannable(c.types[field]) {
+			if pField != field && scannable(c.types[field]) { //if not PK - try to set
```

Comment:
only change to this is enough?
```go
if (pField != field || pField == field && !c.ignorePrimary) && scannable(c.types[field]) {
```

Reply:
and after that we maybe we don't need `mutablePField`?

Reply:
Sounds reasonable, however, mutable serial and bigserial PK does not make sense. Hence, the mutable PK is here to ensure that PK is something other than that.

Reply:
> mutable serial and bigserial PK does not make sense

sorry I don't understand this part

if what you mean is to allow non autogenerated id only for field other than integer (or bigint), then you won't be able to accommodate user that use library like snowflake (go version: https://github.com/sony/sonyflake)
so it's perfectly fine to accept any type as primary, no need to think about whether it's autogenerated or not

Reply:
Makes sense. Actually, what do think about this change in general? Maybe, it worth to just enforce users to set the PK using explicit `rel.Set` than to introduce such complicated logic? Or suggested changes are valid from perspective of library's idea?

Reply:
I think the suggested change is not complicated at all 🤔 :
1. add `ignorePrimaryField` to `Changeset` struct
2. set `ignorePrimaryField` to true when `Convert` is used
3. Update `Apply` to use `if (pField != field || pField == field && !c.ignorePrimary) && scannable(c.types[field]) {` condition

Reply:
Ok then. In general it would be nice to revisit the changeset logic, because it does not allow for composite PK, and maybe some rules about it's mutation should be introduced - again I think that PK can be inserted, but not updated.

Reply:
not sure why composite pk is not allowed, but feel free to open other issue if it still happening after this pr merged